target_disk_unix.go

```diff
@@ -15,10 +15,10 @@ import (
 
 // Chown changes the numeric uid and gid of the named file.
 func (d *TargetDisk) Chown(name string, uid, gid int) error {
-	if os.Geteuid() != 0 {
-		return &fs.PathError{Op: "chown", Path: name, Err: os.ErrPermission}
+	if err := os.Lchown(name, uid, gid); err != nil {
+		return &fs.PathError{Op: "chown", Path: name, Err: err}
```

Comment:
Should this return the error? Maybe...
```suggestion
		return fmt.Errorf("chown failed: %w", err)
```

Also it would be good to have a test case for this error to check we don't silently fail when we shouldn't. You could force the error by using a nonexistent path maybe?

Reply:
Mhm ... I thought it might be nice to wrap the error to have more structure/context, e.g., path/op, but happy to just return the error "straighter".

I will add an explicit test case to verify the error edge-case.

Reply:
Ah, when I read this diff the first time, it looked like it wasn't returning any error at all, now when I see it it seems as though the error was being returned. Sorry for the confusion.

Reply:
> The code path is already tested in [TestUnpackWithPreserveOwnershipAsNonRoot](https://github.com/hashicorp/go-extract/blob/3c7c310952128d3d1dfbc5f547d8f876068569b1/unpack_unix_test.go#L55-L88).

Looking at those [test cases](https://github.com/hashicorp/go-extract/blob/5580e05a86899ffd7c3876c5a94a8572bb2d4a87/unpack_test.go#L1607-L1608), I can't see one where we actively expect an error, so I don't think the coverage is explicit enough to ensure we don't break this behaviour in the future. I will submit a patch to add explicit coverage, you can decide how to use it :)

Reply:
Hey, that sounds great. Looking forward to your patch! let me know when we can sync on it.

Reply:
Hey Jan, still catching up after the break, I can't remember if I created this patch or not, I think I stopped because I realised there were other issues with tests that should be addressed as well. I am going to re-familiarise myself with this PR today, and will let you know once I'm back up to speed!